include/zephyr/net/ieee802154_mgmt.h

```diff
@@ -330,6 +330,8 @@ struct ieee802154_req_params {
 	uint8_t len;
 	/** Link quality information, between 0 and 255 */
 	uint8_t lqi;
+	/** Flag if association is allowed */
+	bool association_allowed;
```

Comment:
Can we call this "assoc_permit", "association_permit" or "association_permitted" please (whatever you prefer), that's how the standard calls it, see section 8.4.3.1, table 8-94 and many other places in which "association permitted" is used.